appengine/pond/duck/js/datastore.js

```diff
@@ -51,27 +79,53 @@ Pond.Duck.Datastore.getAllDucks = function(onLoadCallback) {
  * @param {!Function} onLoadCallback The function to be called when response is received.
  */
 Pond.Duck.Datastore.getDuck = function(duckKey, onLoadCallback) {
-    var url = 'pond-storage/ducks?key='+ duckKey;
+    var url = 'pond-storage/get?key='+ duckKey;
     Pond.Duck.Datastore.makeRequest_(url, 'GET', [], onLoadCallback);
 };
 
 /**
- * TODO: Remove. We should instead be using the commone make request.
+ * TODO: Remove. We should instead be using the common make request.
  */
 Pond.Duck.Datastore.makeRequest_ = function(url, type, data, onLoadCallback) {
     var xhr = new XMLHttpRequest();
     xhr.open(type, url);
-    xhr.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
+    if (type !== 'GET') {
+      xhr.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
+    }
     xhr.onload = onLoadCallback;
+    console.log(url);
     if (data) {
         console.log(data.join('&'));
         xhr.send(data.join('&'));
     } else {
         xhr.send();
     }
 };
-  
-  
+
+/**
+ * Callback function for request that displays error on failure.
+ */
+Pond.Duck.Datastore.messageOnError = function() {
+    if (this.status !== 200) {
+        var text = BlocklyGames.getMsg('Games_httpRequestError') + '\nStatus: '
+            + this.status;
+        BlocklyDialogs.storageAlert(null, text);
+    }
+};
+
+/**
+ * Callback function for request that redirects to new duck
+ */
+Pond.Duck.Datastore.redirectToNewDuck = function() {
+    Pond.Duck.Datastore.messageOnError.call(this);
```

Comment:
Duck board should also redirect to new duck on creation. I put it here since I expect it to be a common callback for requests and would otherwise result in duplicate code.